### PR TITLE
Feature 82 statistics pdf

### PR DESCRIPTION
**Pull request by Jdallos**

### CODE REVIEW

File: src/components/statistics/survey-statistics.tsx [hunk 1]
```diff
@@ -284,6 +379,44 @@ const SurveyStatistics = ({ devices, survey }: Props) => {
...
+    if (
+      !(
+        combinedChartsData?.answerDistributionCharts.length &&
+        combinedChartsData?.popularTimesAndDeviceCharts.length &&
+        surveyStatistics &&
+        renderPdfCharts
+      )
+    ) {
+      return <></>;
+    }
```
Comment (lauriLukkarinen) [inline src/components/statistics/survey-statistics.tsx:395]:
```suggestion
    const { answerDistributionCharts, popularTimesAndDeviceCharts } = combinedChartsData || {};
    const chartsHaveData = answerDistributionCharts?.length && popularTimesAndDeviceCharts?.length;
    const shouldUpdatePdfInstance = surveyStatistics && renderPdfCharts && chartsHaveData;

    if (!shouldUpdatePdfInstance) {
      return;
    }
```

File: src/localization/fi.json [hunk 2]
```diff
@@ -167,6 +167,8 @@
     "editor": "Editori",
     "published": "Julkaistu",
     "statistics": "Tilastot",
+    "pdfDownload": "Lataa PDF",
+    "pdfGenerate": "Luoda pdf",
```
Comment (lauriLukkarinen) [inline src/localization/fi.json:171]:
```suggestion
    "pdfGenerate": "Luo PDF",
```

**Review by lauriLukkarinen — CHANGES_REQUESTED** [review]

File: src/components/statistics/survey-statistics.tsx [hunk 3]
```diff
@@ -94,10 +120,78 @@ const SurveyStatistics = ({ devices, survey }: Props) => {
...
+    if (renderPdfCharts && pdfInstance.url && !pdfInstance.loading && pdfReady) {
+      const link = document.createElement("a");
+      link.href = pdfInstance.url;
+      link.setAttribute("download", `${survey.title.replaceAll(" ", "-")}.pdf`);
+
+      document.body.appendChild(link);
+
+      link.click();
+
+      link.parentNode?.removeChild(link);
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/statistics/survey-statistics.tsx:182]: Done

File: src/components/statistics/pdf-horizontal-chart.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import ChartTooltip from "./chart-tooltip";
+import { Bar, BarChart, ResponsiveContainer, Tooltip, XAxis } from "recharts";
+
+/**
+ * Component properties
+ */
+interface Props {
+  data: any[];
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/components/statistics/pdf-horizontal-chart.tsx:8]: Nope, just copied and pasted from the original horizontal chart. Have now typed pdf and original appropriately

**Review by lauriLukkarinen — APPROVED** [review]

**Pull request merged**